File: Practice(Beginner)/ATM.cpp
// Source: https://www.codechef.com/problems/HS08TEST

#include <bits/stdc++.h>
using namespace std;

int main() {
    int withdraw;
    double balance;

    cin >> withdraw >> balance;

    if((withdraw + 0.5) > balance or withdraw%5 != 0){
        cout << fixed << setprecision(2) << balance;
    }else{
        cout << fixed << setprecision(2) << balance - withdraw - 0.5;
    }
    return 0;
}

File: Practice(Beginner)/Python3/ATM.py
# Source: https://www.codechef.com/problems/HS08TEST

w,a = map(float,input().split())
if(int(w)%5 == 0 and 0 < w < (a-0.50)):
    w += 0.50
else:
    w = 0.0
print('{0:.2f}'.format(a-w))
